src/app/shared/analytics/format-error.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Formats an `ErrorEvent` to a human-readable string that can
+ * be sent to Google Analytics.
+ */
+export function formatErrorForAnalytics(event: ErrorEvent): string {
+  const {message, filename, colno, lineno, error} = event;
+  let stack: string;
+
+  if (error instanceof Error && error.stack) {
+    stack = error.stack
+      // strip the leading "Error: " from the stack trace
+      .replace(/^Error: /, '')
+      // strip the message from the stack trace, if present
+      .replace(error.message + '\n', '')
+      // strip leading spaces
+      .replace(/^ +/gm, '')
+      // strip all leading "at " for each frame
+      .replace(/^at /gm, '')
+      // replace long urls with just the last segment: `filename:line:column`
+      .replace(/(?: \(|@)http.+\/([^/)]+)\)?(?:\n|$)/gm, '@$1\n')
+      // replace "eval code" in Edge
+      .replace(/ *\(eval code(:\d+:\d+)\)(?:\n|$)/gm, '@???$1\n');
```

Comment:
FYI: This logic is taken from `angular.io`. Seems fine to do this, but not sure if strictly needed at this point..